crates/scheduler/src/call/led.rs

```diff
@@ -44,8 +44,10 @@ fn count<B: Board>(call: SchedulerCall<B, api::count::Sig>) {
 fn get<B: Board>(call: SchedulerCall<B, api::get::Sig>) {
     let api::get::Params { led } = call.read();
     let result = try {
-        let id = Id::new(*led as usize).ok_or(Trap)?;
-        board::Led::<B>::get(id)
+        match Id::new(*led as usize) {
```

Comment:
For convenience, we should probably change `Id::new()` to return `Result<Self, Error>`. Then, I would expect this function to look like:

```rust
let result = try {
    board::Led::<B>::get(Id:new(*led as usize)?)
};
call.reply(Ok(result));
```

Note how the reply is `Ok(result)` showing that `led::get()` never traps. (Maybe type annotations are needed though.)

Reply:
Would you like me to change it in this PR? I ask because it seems like I would have to touch other modules given the type change, and perhaps we want to keep this PR isolated.

Reply:
Yes, another PR is probably better. This PR is already improving the situation and only has a small code duplication which is going to be fixed by the other PR.

Reply:
I've created #435 to track that.